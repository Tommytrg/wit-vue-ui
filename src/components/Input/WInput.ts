export enum InputType {
  Base = 'base',
  Action = 'action'
}

export const inputTypes = Object.values(InputType)
