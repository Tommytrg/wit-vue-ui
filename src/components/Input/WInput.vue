<template>
  <div class="flex">
    <input
      :type="contentType"
      :placeholder="placeholder"
      :name="contentType"
      :class="{ [`[&&]:border-black-950 border-2`]: outlined, [`[&&]:rounded-l-full [&&]:rounded-none`]: isActionInput }"
      class="text-black-950 outline-none border-white-50 required email rounded-full border min-w-[300px] sm:min-w-full py-sm px-md block w-full shadow-sm text-sm focus:z-10 focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-neutral-400 dark:placeholder-neutral-500 dark:focus:ring-neutral-600"
      value=""
    />
    <input
      v-if="isActionInput"
      @click="action"
      type="submit"
      :name="actionLabel"
      class="text-wit-blue-500 margin rounded-r-full py-sm px-md inline-flex justify-center items-center gap-x-2 text-sm font-semibold rounded-e-md border border-black-950 bg-black-950 text-white hover:bg-blue-700 disabled:opacity-50 disabled:pointer-events-none cursor-pointer"
      :value="actionLabel"
    />
  </div>
</template>

<script setup lang="ts">
import { PropType, computed } from 'vue'
import { inputTypes, InputType } from './WInput'

const props = defineProps({
  type: {
    type: String as PropType<InputType>,
    default: InputType.Base,
    validator(value: InputType) {
      return inputTypes.includes(value)
    }
  },
  outlined: {
    type: Boolean,
    default: false,
  },
  contentType: {
    type: String,
    required: true
  },
  placeholder: {
    type: String,
    default: null,
  },
  actionLabel: {
    type: String,
    required: false,
  }
})
const emit = defineEmits(['action'])
const isActionInput = computed(() => props.type === InputType.Action)
function action() {
  emit('action')
}
</script>
