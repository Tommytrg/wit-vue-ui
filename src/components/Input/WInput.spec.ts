import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import WInput from './WInput.vue'
import { InputType } from './WInput'

describe('WInput', () => {
  it('base', () => {
    const wrapper = mount(WInput, {
      props: {
        type: InputType.Base,
        contentType: 'string',
        placeholder: 'Amount'
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('action', () => {
    const wrapper = mount(WInput, {
      props: {
        type: InputType.Action,
        contentType: 'email',
        placeholder: 'email@mail.com'
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
