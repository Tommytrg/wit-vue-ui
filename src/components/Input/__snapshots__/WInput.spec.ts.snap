// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WInput > action 1`] = `
<div
  class="flex"
>
  <input
    class="[&&]:rounded-l-full [&&]:rounded-none [&&]:dark:focus:border-r-0 [&&]:focus:border-r-0 text-black-950 dark:text-white-50 bg-white-50 dark:bg-black-950 outline-none border-white-50 dark:border-black-950 required email rounded-full border min-w-[300px] sm:min-w-full py-sm px-md block w-full shadow-sm text-sm focus:z-10 focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-neutral-400 dark:placeholder-neutral-500 focus:border-white-50 dark:focus:border-wit-blue-500 focus:border"
    name="email"
    placeholder="email@mail.com"
    type="email"
    value=""
  />
  <input
    class="text-wit-blue-500 relative z-0 dark:text-black-950 margin rounded-r-full py-sm px-md inline-flex justify-center items-center gap-x-2 text-sm font-semibold rounded-e-md border border-black-950 dark:border-wit-blue-500 bg-black-950 dark:bg-wit-blue-500 text-white hover:bg-blue-700 disabled:opacity-50 disabled:pointer-events-none cursor-pointer"
    type="submit"
  />
</div>
`;

exports[`WInput > base 1`] = `
<div
  class="flex"
>
  <input
    class="text-black-950 dark:text-white-50 bg-white-50 dark:bg-black-950 outline-none border-white-50 dark:border-black-950 required email rounded-full border min-w-[300px] sm:min-w-full py-sm px-md block w-full shadow-sm text-sm focus:z-10 focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-neutral-400 dark:placeholder-neutral-500 focus:border-white-50 dark:focus:border-wit-blue-500 focus:border"
    name="string"
    placeholder="Amount"
    type="string"
    value=""
  />
  <!--v-if-->
</div>
`;
